crates/sui-checkpoint-object-store-indexer/src/lib.rs

```diff
@@ -0,0 +1,323 @@
+                rpc::v2::Checkpoint::path_builder().summary().bcs().value(),
+                rpc::v2::Checkpoint::path_builder().signature().finish(),
+                rpc::v2::Checkpoint::path_builder().contents().bcs().value(),
+                rpc::v2::Checkpoint::path_builder()
+                    .transactions()
+                    .transaction()
+                    .bcs()
+                    .value(),
+                rpc::v2::Checkpoint::path_builder()
+                    .transactions()
+                    .effects()
+                    .bcs()
+                    .value(),
+                rpc::v2::Checkpoint::path_builder()
+                    .transactions()
+                    .effects()
+                    .unchanged_loaded_runtime_objects()
+                    .finish(),
+                rpc::v2::Checkpoint::path_builder()
+                    .transactions()
+                    .events()
+                    .bcs()
+                    .value(),
+                rpc::v2::Checkpoint::path_builder()
+                    .objects()
+                    .objects()
+                    .bcs()
+                    .value(),
+            ])
+            .into()
+        });
+
+        let sequence_number = checkpoint.summary.sequence_number;
+        let proto_checkpoint = rpc::v2::Checkpoint::merge_from(checkpoint.as_ref(), &MASK);
+        let proto_bytes = Bytes::from(proto_checkpoint.encode_to_vec());
+
+        Ok(vec![CheckpointBlob {
+            sequence_number,
+            proto_bytes,
+        }])
+    }
+}
+
+#[async_trait::async_trait]
+impl Handler for CheckpointBlobPipeline {
+    type Store = ObjectStore;
+    type Batch = Option<Self::Value>;
+
+    async fn commit<'a>(
+        &self,
+        batch: &Self::Batch,
+        conn: &mut <Self::Store as StoreTypes>::Connection<'a>,
+    ) -> anyhow::Result<usize> {
+        let Some(blob) = batch else {
+            return Ok(0);
+        };
+
+        let mut path = format!("{}.binpb", blob.sequence_number);
+        let data: Bytes = if let Some(level) = self.compression_level {
+            path = format!("{}.zst", path);
+            tokio::task::spawn_blocking({
+                let bytes = blob.proto_bytes.clone();
+                move || {
+                    let compressed = BytesMut::new();
+                    let mut writer = compressed.writer();
+                    let mut encoder = zstd::Encoder::new(&mut writer, level)?;
+                    std::io::copy(&mut &bytes[..], &mut encoder)?;
+                    encoder.finish()?;
+                    Ok::<Bytes, std::io::Error>(writer.into_inner().freeze())
+                }
+            })
+            .await??
+        } else {
+            blob.proto_bytes.clone()
+        };
+
+        conn.object_store()
+            .put(&ObjectPath::from(path), data.into())
+            .await?;
+        Ok(1)
+    }
+}
+
+pub struct EpochsPipeline;
```

Comment:
minor nit: it would be nice to follow the structure we use for other indexers where we pull these out into separate modules in a `handlers`s module -- just for consistency/familiary.